### DISCUSSION TITLE
Tesla Smart TSL-TRV-TV01ZG schedule (programming)

### DISCUSSION THREAD

**Discussion by dvovla** [Koenkk/zigbee2mqtt#11096] [Devices]
Hello. I am testing the Tesla Smart TSL-TRV-TV01ZG trv and I need to know if and how it is possible to set a schedule. In the Z2M documentation, this trv is listed under TV02-Zigbee. But it is not clear from the description what the payload should look like. Can anyone advise me? Thanks.

**Comment by Mikej0h** [comment]
Not to hi-jack this question, but I'm in the same boat.
I have the TV02-Zigbee (so similar as TS-lister).

The documentation seems incomplete to determine what is able to send as a schedule:
https://www.zigbee2mqtt.io/devices/TV02-Zigbee.html#programming-composite

I've been trying to send as payload:
`{"undefined":{"schedule_monday":"6h:0m 17°C, 12h:0m 17°C, 14h:0m 17°C, 17h:0m 17°C, 24h:0m 17°C , 24h:0m 17°C , 24h:0m 17°C , 24h:0m 17°C , 24h:0m 17°C , 24h:0m 17°C","schedule_tuesday":"6h:0m 17°C, 12h:0m 17°C, 14h:0m 17°C, 17h:0m 17°C, 24h:0m 17°C , 24h:0m 17°C , 24h:0m 17°C , 24h:0m 17°C , 24h:0m 17°C , 24h:0m 17°C","schedule_wednesday":"6h:0m 17°C, 12h:0m 17°C, 14h:0m 17°C, 17h:0m 17°C, 24h:0m 17°C , 24h:0m 17°C , 24h:0m 17°C , 24h:0m 17°C , 24h:0m 17°C , 24h:0m 17°C","schedule_thursday":"6h:0m 17°C, 12h:0m 17°C, 14h:0m 17°C, 17h:0m 17°C, 24h:0m 17°C , 24h:0m 17°C , 24h:0m 17°C , 24h:0m 17°C , 24h:0m 17°C , 24h:0m 17°C","schedule_friday":"6h:0m 17°C, 12h:0m 17°C, 14h:0m 17°C, 17h:0m 17°C, 24h:0m 17°C , 24h:0m 17°C , 24h:0m 17°C , 24h:0m 17°C , 24h:0m 17°C , 24h:0m 17°C","schedule_saturday":"6h:0m 17°C, 12h:0m 17°C, 14h:0m 17°C, 17h:0m 17°C, 24h:0m 17°C , 24h:0m 17°C , 24h:0m 17°C , 24h:0m 17°C , 24h:0m 17°C , 24h:0m 17°C","schedule_sunday":"6h:0m 17°C, 12h:0m 17°C, 14h:0m 17°C, 17h:0m 17°C, 24h:0m 17°C , 24h:0m 17°C , 24h:0m 17°C , 24h:0m 17°C , 24h:0m 17°C , 24h:0m 17°C"}}` (just setting all temps to 17 - although disabled would it even better)

But zigbee2mqtt reports:
No converter available for 'undefined'

Hopefully you can push us in the right direction. Thanks!
